### ISSUE TITLE
1.21.0: IllegalArgumentException: File/directory does not exist: IdentityTransform.transform

### ISSUE THREAD

**Issue by yogurtearl** [detekt/detekt#5150]
In a build using:
Gradle `7.5`
AGP `7.2.1`
Kotlin `1.6.21`
Configuration Cache: ON

**TL;DR** 
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

**If you have an Android module that depends on a Kotlin (non-android) only module and have configuration cache turned on, then you will get an error with detekt 1.21.0**

 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

When upgrading from detekt `1.20.0` to `1.21.0` (or to `main-SNAPSHOT`) with **no other changes**, our android build starts failing with the error mentioned below.

The error only happens when a `detekt` task is on the execution list and only with `1.21.0` (or latest SNAPSHOT). Same error with `1.21.0-RC1` as well. 
AND it only happens with configuration cache turned on, if I turn off configuration cache the error goes away.

If I re-run gradle a second time the error goes away. 
but if I do `rm /Users/me/project/lib1/build/libs/lib1.jar` and  then re-run again, the error comes back. 

This is 100% reproducible within our build with `1.21.0`, but never happens with `1.20.0`. 

From the detekt bits in the stack trace and looking at the code, I think this change is the cause: https://github.com/detekt/detekt/pull/4762

Detekt in the stack trace below:
```
        at io.gitlab.arturbosch.detekt.invoke.ClasspathArgument.toArgument(CliArgument.kt:46)
        at io.gitlab.arturbosch.detekt.Detekt.arguments$lambda-1(Detekt.kt:234)
```

Calling `toArgument` on `ClasspathArgument` inside a `provider` doesn't play well with config cache, because config cache runs the `IdentityTransform` from AGP before the `jar` that it is trying to transform has build built. 

The key aspects are likely:
-  android `lib2` that depends on non-android kotlin only `lib1`
-  Config Cache turned on
- `rm lib1/build lib2/build` before the build
- call the `detektMain` task on all the projects

```
* What went wrong:
Execution failed for task ':lib2:kaptGenerateStubsDebugKotlin'.
> Error while evaluating property 'filteredArgumentsMap' of task ':lib2:kaptGenerateStubsDebugKotlin'
   > Could not resolve all files for configuration ':lib2:debugCompileClasspath'.
      > Failed to transform lib1.jar (project lib1) to match attributes {artifactType=android-classes-jar, org.gradle.category=library, org.gradle.dependency.bundling=external, org.gradle.jvm.environment=standard-jvm, org.gradle.jvm.version=8, org.gradle.libraryelements=jar, org.gradle.usage=java-api, org.jetbrains.kotlin.platform.type=jvm}.
         > Execution failed for IdentityTransform: /Users/me/project/lib1/build/libs/lib1.jar.
            > File/directory does not exist: /Users/me/project/lib1/build/libs/lib1.jar

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':lib2:kaptGenerateStubsDebugKotlin'.
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:38)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
 
Caused by: org.gradle.api.internal.tasks.properties.PropertyEvaluationException: Error while evaluating property 'filteredArgumentsMap' of task ':lib2:kaptGenerateStubsDebugKotlin'
        at org.gradle.api.internal.tasks.properties.InputParameterUtils.prepareInputParameterValue(InputParameterUtils.java:33)
        at org.gradle.api.internal.tasks.execution.TaskExecution.lambda$visitRegularInputs$1(TaskExecution.java:309)
        at org.gradle.internal.execution.fingerprint.impl.DefaultInputFingerprinter$InputCollectingVisitor.visitInputProperty(DefaultInputFingerprinter.java:106)
        at org.gradle.api.internal.tasks.execution.TaskExecution.visitRegularInputs(TaskExecution.java:309)
        at org.gradle.internal.execution.fingerprint.impl.DefaultInputFingerprinter.fingerprintInputProperties(DefaultInputFingerprinter.java:61)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.captureExecutionStateWithOutputs(CaptureStateBeforeExecutionStep.java:193)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.lambda$captureExecutionState$1(CaptureStateBeforeExecutionStep.java:141)
        at org.gradle.internal.execution.steps.BuildOperationStep$1.call(BuildOperationStep.java:37)
 
Caused by: org.gradle.api.internal.artifacts.ivyservice.DefaultLenientConfiguration$ArtifactResolveException: Could not resolve all files for configuration ':lib2:debugCompileClasspath'.
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.rethrowFailure(DefaultConfiguration.java:1524)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.access$3700(DefaultConfiguration.java:159)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration$DefaultResolutionHost.rethrowFailure(DefaultConfiguration.java:2174)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration$ConfigurationFileCollection.visitContents(DefaultConfiguration.java:1496)
 
Caused by: org.gradle.api.internal.artifacts.transform.TransformException: Failed to transform lib1.jar (project lib1) to match attributes {artifactType=android-classes-jar, org.gradle.category=library, org.gradle.dependency.bundling=external, org.gradle.jvm.environment=standard-jvm, org.gradle.jvm.version=8, org.gradle.libraryelements=jar, org.gradle.usage=java-api, org.jetbrains.kotlin.platform.type=jvm}.
        at org.gradle.api.internal.artifacts.transform.TransformedProjectArtifactSet.visit(TransformedProjectArtifactSet.java:118)
        at org.gradle.api.internal.artifacts.ivyservice.resolveengine.artifact.ParallelResolveArtifactSet$VisitingSet$StartVisitAction.visitResults(ParallelResolveArtifactSet.java:100)
        at org.gradle.api.internal.artifacts.ivyservice.resolveengine.artifact.ParallelResolveArtifactSet$VisitingSet.visit(ParallelResolveArtifactSet.java:69)
        at org.gradle.api.internal.artifacts.ivyservice.DefaultLenientConfiguration.visitArtifacts(DefaultLenientConfiguration.java:301)
        at org.gradle.api.internal.artifacts.ivyservice.DefaultLenientConfiguration.access$600(DefaultLenientConfiguration.java:75)
        at org.gradle.api.internal.artifacts.ivyservice.DefaultLenientConfiguration$3.run(DefaultLenientConfiguration.java:249)

Caused by: org.gradle.api.internal.artifacts.transform.TransformException: Execution failed for IdentityTransform: /Users/me/project/lib1/build/libs/lib1.jar.
        at org.gradle.api.internal.artifacts.transform.DefaultTransformerInvocationFactory$1.lambda$mapResult$3(DefaultTransformerInvocationFactory.java:159)
        at org.gradle.internal.Try$Failure.mapFailure(Try.java:273)
        at org.gradle.api.internal.artifacts.transform.DefaultTransformerInvocationFactory$1.mapResult(DefaultTransformerInvocationFactory.java:159)
        at org.gradle.api.internal.artifacts.transform.DefaultTransformerInvocationFactory$1.processCachedOutput(DefaultTransformerInvocationFactory.java:145)
        at org.gradle.api.internal.artifacts.transform.DefaultTransformerInvocationFactory$1.processCachedOutput(DefaultTransformerInvocationFactory.java:142)

Caused by: java.lang.IllegalArgumentException: File/directory does not exist: /Users/me/project/lib1/build/libs/lib1.jar
        at com.android.build.gradle.internal.dependency.IdentityTransform.transform(IdentityTransform.kt:61)
        at org.gradle.api.internal.artifacts.transform.DefaultTransformer.transform(DefaultTransformer.java:264)
        at org.gradle.api.internal.artifacts.transform.DefaultTransformerInvocationFactory$AbstractTransformerExecution$1.call(DefaultTransformerInvocationFactory.java:290)
        at org.gradle.api.internal.artifacts.transform.DefaultTransformerInvocationFactory$AbstractTransformerExecution$1.call(DefaultTransformerInvocationFactory.java:285)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)


    .
    .
    .
        at org.gradle.api.internal.file.collections.UnpackingVisitor.add(UnpackingVisitor.java:64)
        at org.gradle.api.internal.file.collections.DefaultConfigurableFileCollection$UnresolvedItemsCollector.visitContents(DefaultConfigurableFileCollection.java:372)
        at org.gradle.api.internal.file.collections.DefaultConfigurableFileCollection.visitChildren(DefaultConfigurableFileCollection.java:284)
        at org.gradle.api.internal.file.CompositeFileCollection.visitContents(CompositeFileCollection.java:119)
        at org.gradle.api.internal.file.AbstractFileCollection.getFiles(AbstractFileCollection.java:129)
        at org.gradle.api.internal.file.AbstractFileCollection.iterator(AbstractFileCollection.java:175)
        at kotlin.collections.CollectionsKt___CollectionsKt.joinTo(_Collections.kt:3344)
        at kotlin.collections.CollectionsKt___CollectionsKt.joinToString(_Collections.kt:3364)
        at kotlin.collections.CollectionsKt___CollectionsKt.joinToString$default(_Collections.kt:3363)
        at io.gitlab.arturbosch.detekt.invoke.ClasspathArgument.toArgument(CliArgument.kt:46)
        at io.gitlab.arturbosch.detekt.Detekt.arguments$lambda-1(Detekt.kt:234)
        at org.gradle.api.internal.provider.DefaultProvider.calculateOwnValue(DefaultProvider.java:72)
        at org.gradle.api.internal.provider.AbstractMinimalProvider.calculateExecutionTimeValue(AbstractMinimalProvider.java:142)
        at org.gradle.configurationcache.serialization.codecs.FixedValueReplacingProviderCodec.encodeProvider(ProviderCodecs.kt:72)
        at org.gradle.configurationcache.serialization.codecs.ProviderCodec.encode(ProviderCodecs.kt:135)
        at org.gradle.configurationcache.serialization.codecs.ProviderCodec.encode(ProviderCodecs.kt:128)
        at org.gradle.configurationcache.serialization.codecs.BindingsBackedCodec.encode(BindingsBackedCodec.kt:52)
        at org.gradle.configurationcache.serialization.DefaultWriteContext.write(Contexts.kt:85)
```

**Comment by 3flex** [comment]
It's very weird because it's a kapt task failing, not a detekt task.

Any chance you can try Kotlin 1.7.10 in your project, with detekt snapshot? Might not help, but there are always fixes being made for things like config cache.

A build scan might be helpful. A reproducer would be even more helpful.

**Comment by yogurtearl** [comment]
@3flex I was able to repro with a fairly simple unit test, see https://github.com/detekt/detekt/pull/5151

If you have an Android module that depends on a Kotlin (non-android) only module and have configuration cache turned on, then you will get this error with detekt 1.21.0 (and on `main`)

**Comment by yogurtearl** [comment]
If I do `git revert 950554c173cbe09c201107194ca7c2a6f2766549` (which reverts https://github.com/detekt/detekt/pull/4762 ) then the unit test in https://github.com/detekt/detekt/pull/5151 passes.

**Comment by ILikeYourHat** [comment]
Any update? I'm stuck on Detekt 1.20.0 because of this
